In [5]:
import numpy as np
import pandas as pd
from sklearn.datasets import load_iris
iris = load_iris()

Kode ini digunakan untuk mengimpor library yang diperlukan dan memuat dataset Iris dari Scikit-learn untuk digunakan dalam proses klasifikasi.

In [6]:
df = pd.DataFrame(iris.data, columns=iris.feature_names)
df["Species"] = iris.target_names[iris.target]
X = df.drop(columns=["Species"])
y = df["Species"]
print(X)

     sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                  5.1               3.5                1.4               0.2
1                  4.9               3.0                1.4               0.2
2                  4.7               3.2                1.3               0.2
3                  4.6               3.1                1.5               0.2
4                  5.0               3.6                1.4               0.2
..                 ...               ...                ...               ...
145                6.7               3.0                5.2               2.3
146                6.3               2.5                5.0               1.9
147                6.5               3.0                5.2               2.0
148                6.2               3.4                5.4               2.3
149                5.9               3.0                5.1               1.8

[150 rows x 4 columns]


Kode ini digunakan untuk **mengubah dataset Iris menjadi DataFrame, menambahkan label species, serta memisahkan data menjadi fitur (X) dan target (y) untuk proses klasifikasi.

In [7]:
y

0         setosa
1         setosa
2         setosa
3         setosa
4         setosa
         ...    
145    virginica
146    virginica
147    virginica
148    virginica
149    virginica
Name: Species, Length: 150, dtype: str

Output ini menunjukkan label species pada dataset Iris, yaitu setosa, versicolor, dan virginica, dengan total 150 data.

In [8]:
from sklearn.preprocessing import LabelEncoder
lab = LabelEncoder()
y = lab.fit_transform(y)
y

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

Kode ini digunakan untuk mengubah label species dari bentuk teks menjadi angka, yaitu setosa = 0, versicolor = 1, dan virginica = 2, agar dapat diproses oleh model.

In [9]:
from sklearn.model_selection import train_test_split
X_train, X_test = train_test_split(X, test_size=0.2, random_state=0)
y_train, y_test = train_test_split(y, test_size=0.2, random_state=0)

Kode ini digunakan untuk membagi dataset menjadi data training sebesar 80% dan data testing sebesar 20% untuk melatih dan menguji model.

In [10]:
from sklearn.naive_bayes import GaussianNB
model = GaussianNB()
NB_model = model.fit(X_train, y_train)

Kode ini digunakan untuk membuat model Gaussian Naïve Bayes dan melatihnya menggunakan data training (X_train dan y_train).

In [12]:
result = NB_model.predict(X_test)
result

array([2, 1, 0, 2, 0, 2, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 2, 1,
       0, 0, 2, 0, 0, 1, 1, 0])

Kode ini digunakan untuk memprediksi species pada data testing menggunakan model Gaussian Naïve Bayes yang sudah dilatih.

In [ ]:
from sklearn.metrics import accuracy_score
print('Accuracy of data testing: ', accuracy_score(y_test, result)*100)

Accuracy of data testing:  96.66666666666667


Kode ini digunakan untuk menghitung akurasi model pada data testing, dengan hasil akurasi sebesar 96,67%.

In [14]:
input_data = [[5.1, 3.5, 1.4, 0.2]]
result = NB_model.predict(input_data)
prediction_label = lab.inverse_transform(result)
print('Hasil prediksi untuk input baru adalah:', prediction_label)

Hasil prediksi untuk input baru adalah: ['setosa']


c:\Users\Lenovo\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2827: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(


Kode ini digunakan untuk memprediksi species dari data baru, dengan hasil prediksi yaitu setosa. Warning muncul karena input tidak memiliki nama fitur yang sama dengan data training.

In [15]:
input_data = pd.DataFrame(np.array([[5.1, 3.5, 1.4, 0.2]]), columns=iris.feature_names)
result = NB_model.predict(input_data)
prediction_label = lab.inverse_transform(result)
print('Hasil prediksi untuk input baru adalah:', prediction_label)

Hasil prediksi untuk input baru adalah: ['setosa']


Kode ini digunakan untuk memprediksi species data baru dengan format fitur yang sesuai dengan model, sehingga hasil prediksinya adalah setosa tanpa warning.

In [16]:
#TUGAS
input_data = pd.DataFrame(np.array([[7.3, 3.8, 4.6, 2.3]]),columns=iris.feature_names)
result = NB_model.predict(input_data)
prediction_label = lab.inverse_transform(result)
print("Data baru:", [7.3, 3.8, 4.6, 2.3])
print("Hasil prediksi:", prediction_label)


Data baru: [7.3, 3.8, 4.6, 2.3]
Hasil prediksi: ['virginica']


Kode tersebut digunakan untuk memprediksi species bunga Iris berdasarkan data baru [7.3, 3.8, 4.6, 2.3] menggunakan model Gaussian Naïve Bayes, kemudian hasil prediksi diubah dari angka menjadi nama species.
